From 09322e74a4f51d657e850eaeef78209b50a7ee89 Mon Sep 17 00:00:00 2001
From: Javier Maestro <jjmaestro@ieee.org>
Date: Sat, 11 Apr 2026 00:28:05 +0000
Subject: [PATCH] patch: work around the flex 2.6.4 --noline bugs in pgflex

Flex has bugs where --noline doesn't suppress all the line directives
(e.g. https://github.com/westes/flex/pull/704) so, until it's accepted,
we filter them on our side.
---
 src/tools/pgflex | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/tools/pgflex b/src/tools/pgflex
index 65cc92c26d..494fd2e685 100755
--- a/src/tools/pgflex
+++ b/src/tools/pgflex
@@ -84,4 +84,15 @@ if args.fix_warnings:
     if sp.returncode != 0:
         sys.exit(sp.returncode)
 
+# fix flex <= 2.6.4 --noline bugs (e.g. https://github.com/westes/flex/pull/704)
+import re
+
+RE_LINE_DIRECTIVE = re.compile(r'#line \d+ ".*"')
+
+with open(args.output_file) as f:
+    output_lines = f.readlines()
+
+with open(args.output_file, "w") as f:
+    f.writelines(l for l in output_lines if not RE_LINE_DIRECTIVE.match(l))
+
 sys.exit(0)
